### PR TITLE
Cluster and cluster host deregistration events/commands

### PR DESCRIPTION
**Pull request by rtorrero**
# Description

Cluster and cluster host deregistration events and commands

Note: I rebased `deregistration` on `main` before splitting the branch for this PR to keep the development close with current `main`

### CODE REVIEW

File: lib/trento/domain/cluster/commands/deregister_cluster_host.ex [hunk 1]
```diff
@@ -0,0 +1,13 @@
+defmodule Trento.Domain.Commands.DeregisterClusterHost do
+  @moduledoc """
+    Deregister a cluster host
```
Comment (jamie-suse) [inline lib/trento/domain/cluster/commands/deregister_cluster_host.ex:3]: Personally, I always found this wording a bit confusing as it sounds as if a "_cluster host_" is a special type of host.

What about something like "Unregister a host from a cluster" or "Remove a host from a cluster"?

File: lib/trento/domain/cluster/events/cluster_deregistered.ex [hunk 2]
```diff
@@ -0,0 +1,12 @@
+defmodule Trento.Domain.Events.ClusterDeregistered do
+  @moduledoc """
+    This event is emitted when a cluster is decommissioned.
```
Comment (arbulu89) [inline lib/trento/domain/cluster/events/cluster_deregistered.ex:3]: Why the word `decommissioned`?
We use `Deregisted` all over the code. If you want to have the same approach than in host, you could put in parenthesis

File: lib/trento/domain/cluster/commands/deregister_cluster_host.ex [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  """
+  @required_fields :all
+
+  use Trento.Command
+
+  defcommand do
+    field :host_id, Ecto.UUID
+    field :cluster_id, Ecto.UUID
```
Comment (arbulu89) [inline lib/trento/domain/cluster/commands/deregister_cluster_host.ex:11]: Shouldn't we have here `deregistered_at` field?
At the end we will cascade this value from events to commands in the process manager

Comment (CDimonaco) [reply lib/trento/domain/cluster/commands/deregister_cluster_host.ex:11]: +1

File: lib/trento/domain/cluster/commands/deregister_cluster_host.ex (hunk 1 above)
Comment (rtorrero) [reply lib/trento/domain/cluster/commands/deregister_cluster_host.ex:3]: Though I'm not a huge fan of the "deregister" word myself, I'm all about commitment :smile: ... Let's go for now with that word and change it later. I'll make the second part of the sentence more clear though, I think you are right.

File: lib/trento/domain/cluster/events/host_removed_from_cluster.ex [hunk 4]
```diff
@@ -0,0 +1,13 @@
+defmodule Trento.Domain.Events.HostRemovedFromCluster do
+  @moduledoc """
+    This event is emitted when a host is removed from a cluster.
```
Comment (arbulu89) [inline lib/trento/domain/cluster/events/host_removed_from_cluster.ex:3]: I would remove this initial trailling tab, as we don't have it in others

**Review by arbulu89 — APPROVED** [review]
:rocket:

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**